contracts/governancestaking/GovarnanceStaking.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.7.0;
+
+import "../Interfaces.sol";
+
+import "../DAOStackInterfaces.sol";
+import "../utils/NameService.sol";
+
+import "./GovernanceStakingToken.sol";
+
+/**
+ * @title Staking contract that donates earned interest to the DAO
+ * allowing stakers to deposit Tokens
+ * or withdraw their stake in Tokens
+ * the contracts buy intrest tokens and can transfer the daily interest to the  DAO
+ */
+contract GovernanceStaking is GovernanceStakingToken,Initializable{
```

Comment:
governancestakingtoken seems like an erc20.
lets remove it, and make GovernanceStaking implement openzeppelin erc20 contracts
something like upgradablemintableerc20
overwrite all required transfer stuff here

Reply:
replaced